aten/src/ATen/native/cuda/MultiMarginLoss.cu

```diff
@@ -150,6 +151,12 @@ void multi_margin_loss_shape_check(
         target.dim() <= 1 && target.numel() == nframe,
         "inconsistent target size, expected ", nframe, " but got ",
         target.sizes());
+    if (weight && weight->defined()) {
+      TORCH_CHECK(
+          weight->dim() <= 1 && weight->numel() == dim,
+          "inconsistent weight size, expected ", dim, " but got ",
+          weight->sizes());
+    }
```

Comment:
From the docs: "a manual rescaling weight given to each class. If given, it has to be a Tensor of size C. Otherwise, it is treated as if having all ones." But `target` here can be a scalar, so could it be useful to allow scalars for `weight`?

https://pytorch.org/docs/stable/generated/torch.nn.MultiMarginLoss.html